src/CityOnPlanet.cpp

```diff
@@ -198,64 +266,111 @@ CityOnPlanet::~CityOnPlanet()
+	const double pop = 1000000000 * planet->GetSystemBody()->GetPopulation();	// GetPopulation() returns population in whole billions, as in 0.5 -> 500M
+	const int atmo = planet->GetSystemBody()->HasAtmosphere() ? config_atmo_size : config_airless_size;
 
 	Random rand;
 	rand.seed(seed);
 
-	int population = planet->GetSystemBody()->GetPopulation();
+	const vector3d mx = m * vector3d(1, 0, 0);
+	const vector3d mz = m * vector3d(0, 0, 1);
+	citybuildinglist_t *buildings = &s_buildingList;
+
+	const int cellsize_i = 50;
+	const double cellsize = double(cellsize_i); // current widest building = library
 	int cityradius;
 
-	if (planet->GetSystemBody()->HasAtmosphere()) {
-		population *= 1000;
-		cityradius = (population < 200) ? 200 : ((population > START_SEG_SIZE) ? START_SEG_SIZE : population);
+	if (pop > 1000000000) {
+		cityradius = atmo + rand.Int32(30);			// 1 Billion+ cityradius 60..90 (atmospheric)
+	} else if (pop > 200000000) {
+		cityradius = atmo * .8 + rand.Int32(15);	// 200M+ 48..63
+	} else if (pop > 20000000) {
+		cityradius = atmo * .6 + rand.Int32(14);	// 20M+  36..50
+	} else if (pop > 1000000) {
+		cityradius = atmo * .4 + rand.Int32(12);	// 1M+   24..38
 	} else {
-		population *= 100;
-		cityradius = (population < 250) ? 250 : ((population > START_SEG_SIZE_NO_ATMO) ? START_SEG_SIZE_NO_ATMO : population);
+		cityradius = atmo * .2 + rand.Int32(10);	// <1M   12..22
 	}
 
-	citybuildinglist_t *buildings = &s_buildingList;
-	vector3d cent = p;
-	const int cellsize_i = 80;
-	const double cellsize = double(cellsize_i); // current widest building = 92
-	const double bodyradius = planet->GetSystemBody()->GetRadius(); // cache for bodyradius value
-
-	static const int gmid = (cityradius / cellsize_i);
-	static const int gsize = gmid * 2;
+	if (cityradius > 98)
+		cityradius = 98;
+	if (cityradius < 10)
+		cityradius = 10;
 
-	assert((START_SEG_SIZE / cellsize_i) < 100);
-	assert((START_SEG_SIZE_NO_ATMO / cellsize_i) < 100);
-	uint8_t cellgrid[200][200];
-	std::memset(cellgrid, 0, sizeof(cellgrid));
+	// Clear grid occupancy bitfield
+	std::memset(citybits, 0, sizeof(citybits));
 
 	// calculate the size of the station model
-	const int x1 = floor(aabb.min.x / cellsize);
-	const int x2 = ceil(aabb.max.x / cellsize);
-	const int z1 = floor(aabb.min.z / cellsize);
-	const int z2 = ceil(aabb.max.z / cellsize);
-
-	// Clear the cells where the station is
-	for (int x = 0; x <= gsize; x++) {
-		for (int z = 0; z <= gsize; z++) {
-			const int zz = z - gmid;
-			const int xx = x - gmid;
-			if (zz > z1 && zz < z2 && xx > x1 && xx < x2)
-				cellgrid[x][z] = 1;
+	const int station_x1 = floor(aabb.min.x / cellsize) + citygridmidpoint + 1;
+	const int station_x2 = ceil(aabb.max.x / cellsize) + citygridmidpoint;
+	const int station_z1 = floor(aabb.min.z / cellsize) + citygridmidpoint + 1;
+	const int station_z2 = ceil(aabb.max.z / cellsize) + citygridmidpoint;
+
+	if (stationmodelname == "ground_station") {
```

Comment:
Nasty hardcoding this, if we must have tight bounding around certain stations then it prefer there was a list of them in either the config or another json file to control them so that future stations can use it without hardcoding these values

Reply:
Can't it be defined in the station.jsons themselves for fexibility?

Reply:
Id like to leave it in until I can figure out the best way to implement it. It would also touch upon more src files to accomplish. Even if I drop dead tomorrow, the hardcoded example shows what can be done, and how.